### ISSUE TITLE
Add option to "clear" keyspace

### ISSUE THREAD

**Issue by KeKs0r** [rocicorp/replicache#600]
I have the case, where I upgraded the "structure" of a prefix. For this example lets say the structure of my elements under `events/:eventId` has changed.  The issue is, I have not "tracked" every individual event that I sent to the client. So I can't send a `put` for all of them. I would like to "clear" all events, so the client is missing the data and will subsequently send a mutation to read them to the data space.
So allowing the `clear` operation to also have a prefix, to only clear a part of the client store would be helpful.

**Comment by aboodman** [comment]
Can you say more about why it is difficult to know which events the client has? That doesn't match my mental model of how to use Replicache. Is it because it is expensive to learn all the events the user has access to from your data source, or is it because of paging or something?

**Comment by KeKs0r** [comment]
I don't track what is synced on the event level. I am tracking which "Leads" and "Accounts" are synced to the client. And then send all events for the synced leads down to the client. 

And one level deeper why I am doing it?
I think mainly because I thought that level of granularity is sufficient for my use case of providing the right data to the client. And the amount of events can grow quite big (over time 10k+) and not needing to track them, was making the implementation just easier.

**Comment by aboodman** [comment]
But I mean theoretically, at the time you want to update, couldn't you load all the events the user has access to and iterate them? Is it just that this is slow?

It's obviously easy for us to implement this, just trying to understand your usecase better.

(sorry for delayed response)

**Comment by aboodman** [comment]
@arv what are your thoughts on implementing:

```
{ op: "clear", prefix: "foo/" }
```

... or some such. I wonder if we would also want it in the programmatic API to match.

**Comment by arv** [comment]
We should also have a prefix param for the method then.

**Comment by KeKs0r** [comment]
> But I mean theoretically, at the time you want to update, couldn't you load all the events the user has access to and iterate them? Is it just that this is slow?
> 
> It's obviously easy for us to implement this, just trying to understand your usecase better.
> 
> (sorry for delayed response)

You are right, I could do that query. 
The result of that query is state that might have been build up over a long time on the client and could therefore be quite large. Potentially leading me having to paginate several roundtrips to the salesforce api. 

I did solve it with a manual migration for now (running that very slow query, luckily most of my users did not have enormous client state)

I did run a manual migration of the state for now. So this is not block
